widget/widget_renderer.go

```diff
@@ -0,0 +1,34 @@
+package widget
```

Comment:
There aren't any other _renderer.go files in widget, typically the renderer is in the same file as the widget. I actually prefer it being in a separate file, but for consistency should this be moved into widget.go?

Reply:
I prefer to have a separate file because the shadowingRenderer is not bound to the baseWidget. Even the proposed baseRenderer is not.
Specific renderers of a specific widget are another kind of story.
However, I would prefer widget specific packages where the widget and the renderer would reside in different files. But this is not only my personal preference but would also be a breaking change.

Reply:
Makes sense. Yeah that would be my preference too